Paco/src/com/pacoapp/paco/model/ExperimentProviderUtil.java

```diff
@@ -886,6 +906,7 @@ private Event findEventBy(String select, String[] selectionArgs, String sortOrde
       if (cursor != null) {
         cursor.close();
       }
+      readLock.unlock();
```

Comment:
I'm not sure I understand what you mean: should the `unlock()` not happen in the `finally{}` block here?
In the new commit, this code is scrapped in favor of having locks in the `insert/findEvent()` methods, but the `unlock()`s happen in the `finally{}` blocks there too. Let me know if you want me to change that.